sentiment-analysis/USER_GUIDE.md

```diff
@@ -0,0 +1,235 @@
+    
+- consumer_secret:
+    - Description: Consumer secret of your twitter developer account
+    - Type: Required string.
+    - Example: '5ROv9lN1WBfmFtJKmNU4m0hNyFu6J1GB3eSFMbJMwBbmj'.
+
+- access_token:
+    - Description: Token secret of your twitter developer account
+    - Type: Required string.
+    - Example: '5ROv9lN1WBfmFtJKmNU4m0hNyFu6J1GB3eSFMbJMwBbmj'.
+
+- token_secret:
+    - Description: Token secret of your twitter developer account
+    - Type: Required string.
+    - Example: '5ROv9lN1WBfmFtJKmNU4m0hNyFu6J1GB3eSFMbJMwBbmj'.
+  
+- product:
+    - Description: Twitter product label.
+    - Type: Required string.
+    - Example: '30day' or 'fullarchive'.
+    
+- environment:
+    - Description: Environment label of the created on twitter developer account. 
+    - Type: Required string field.
+    - Example: 'development'.
+    
+- query: 
+    - Description: Query used to capture messages. 
+    - Type: Required string field. 
+    - Example: 'Donaldo Trump OR Hilary Cliont OR North Korea'
+    
+- messages_per_request -> 
+    - Description: Number of messages per request.
+    - Type: optional int, max of 100 for sandbox accounts and 500 for paid accounts. This Twitter information, please search for "Twitter Api Prices"
+    - Example: 100
+    - Note: If you don't send a value, twitter will assume default value for your type of account.
+    
+- max_requests_limit:
+    - Description: number of http requests on api
+    - Type: Optional int 10
+    - Example: 10, 
+    - Note: If you don't send a value, you will continue to request data until your account limit expires.  
+    
+- msg_limit:
+    - Description: You can limit your request by message number
+    - Type: Optional int
+    - Example: 1250
+    - Note: If you don't send a value, you will continue to request data until your account limit expires.
+    
+- time_limit:
+    - Description: You can limit your request by time in seconds
+    - Type: optional int
+    - Example: 600 seconds or 10 minutes. 
+    - Note: If you don't send a value, you will continue to request data until your account limit expires.
+    
+- from_date:
+    - Description: Initial date range
+    - Type: required date string
+    - Example: '20181201235959'
+    - Note: Pattern yyyymmddhhmmss
+    
+- to_date:
+    - Description: Final date range
+    - Type: required date string
+    - Example: '20181231235959'
+    - Note: -> Pattern yyyymmddhhmmss 
+
+- languages:
+    - Description: Selected language to be use on querying data
+    - Type: optional string
+    - Example: 'en' or 'en,pt,es'
+    - Note:
+
+JSON example:
+```
+{
+    "credentials":{
+        "consumer_key":"5ROv9lN1WBfmFtJKmNU4m0hNyFu6J1GB3eSFMbJMwBbmj",
+        "consumer_secret":"5ROv9lN1WBfmFtJKmNU4m0hNyFu6J1GB3eSFMbJMwBbmj",
+        "access_token":"5ROv9lN1WBfmFtJKmNU4m0hNyFu6J1GB3eSFMbJMwBbmj",
+        "token_secret":"5ROv9lN1WBfmFtJKmNU4m0hNyFu6J1GB3eSFMbJMwBbmj"
```

Comment:
Are we allowed to use the Twitter API this way? Usually to do work on behalf of a twitter account you need to set up an OAuth flow. Getting the developer to give us their credentials is very dodgy.

Reply:
Actually, we have two contexts for this service:
- Final user service published in the marketplace 
- Service Interface, where this service can be consumed by another service which has the same service owner.

For this moment we are working on the second context, when we are working on the first one, for sure, we should not ask for these parameters and probably we must have another version omitting those parameters.

Reply:
Ok, I think even in case 2, the service should either be preconfigured with the credentials or be passing OAuth tokens which are less dangerous to share. The core credentials shouldn't be passed with every request. As far as I know we don't use SSL either yet, so anyone on the network can see these these secrets.

However, that's a different issue to the fixes you are changing in this PR, so I will merge things now. Can you create an issue to investigate this further at some point?

Reply:
Sure